hera_sim/simulate.py

```diff
@@ -0,0 +1,335 @@
+
+    def _read_data(self, filename, polarizations, **kwargs):
+        uv = UVData()
+        uv.read(filename, polarizations=polarizations, read_data=True, **kwargs)
+        return uv
+
+    def write_data(self, filename, file_type=None, **kwargs):
+        """
+        Write current UVData object to file.
+
+        Args:
+            filename (str): filename to write to. Suffix of filename will determine the type of file to be written,
+                unless `file_type` is set.
+            file_type: (str): one of "miriad", "uvfits" or "uvh5" (i.e. any of the supported write methods of
+                :class:`pyuvdata.UVData`) which determines which write method to call.
+            **kwargs: keyword arguments sent directly to the write method chosen.
+        """
+        if file_type is None:
+            try:
+                file_type = filename.split(".")[-1]
+            except IndexError:
+                raise ValueError(
+                    "if no file_type is given, the filename must be suffixed with either .miriad, .uvfits or .uvh5"
+                )
+
+        if file_type not in ["miriad", "uvfits", "uvh5"]:
+            raise ValueError("file_type must be one of 'miriad', 'uvfits' or 'uvh5'")
+
+        getattr(self.data, "write_%s" % filename.split(".")[-1])(filename, **kwargs)
+
+    def _iterate_baselines(self, with_conj=True):
+        """
+        Iterate through baselines in the data object
+        """
+        for i, ant1 in enumerate(self.data.antenna_numbers):
+            for j, ant2 in enumerate(
+                self.data.antenna_numbers[(i + 1 if with_conj else 0) :]
+            ):
+                # Get the Nblts indices of this baseline (and its conjugate)
+                yield ant1, ant2, self.data.antpair2ind(
+                    ant1, ant2, ordered=not with_conj
+                )
+
+    @_model()
+    def add_eor(self, model, **kwargs):
+        """
+        Add an EoR-like model to the visibilities.
+
+        Args:
+            model (str or callable): either a string name of a model function existing in :mod:`~hera_sim.eor`, or
+                a callable which has the signature ``fnc(lsts, fqs, bl_len_ns, **kwargs)``.
+            **kwargs: keyword arguments sent to the EoR model function, other than `lsts`, `fqs` and `bl_len_ns`.
+        """
+        for ant1, ant2, ind in self._iterate_baselines():
+            lsts = self.data.lst_array[ind]
+            bl_len_m = self.data.uvw_array[ind][
+                0, 0
+            ]  # just the E-W baseline length at this point.
+
+            vis = model(
+                lsts=lsts,
+                fqs=self.data.freq_array[0]
+                * 1e-9,  # Axis 0 is spectral windows, of which at this point there are always 1.
+                bl_len_ns=bl_len_m * 1e9 / 3e8,
+                **kwargs
+            )
+
+            self.data.data_array[
+                ind, 0, :, 0
+            ] += vis  # TODO: not sure about only using first pol.
```

Comment:
See comment above for how to use the proper polarization index. At some point we need to expose a `polarization` kwarg in `model`, but let's worry about that once we have models that actually take a polarization..

Reply:
This is also fixed (for now) in 5e528d5a9505e6d5df19ae9fd57c424265a3d5c4